### PR TITLE
parsers: Add Fortran parser

### PR DESCRIPTION
**Pull request by theHamsta**
Fixes: https://github.com/nvim-treesitter/nvim-treesitter/issues/1194

Previous requests: @oponkork he also created the queries.
- https://nvim-treesitter.zulipchat.com/#narrow/stream/252271-general/topic/Meeting.20everyone/near/219174700

The parser needs an update for the new tree-sitter ABI.

### CODE REVIEW

File: queries/fortran/highlights.scm [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+  "endsubroutine"
+  "enumerator"
+  "equivalence"
+  "exit"
+  "format"
+  "function"
+  "goto"
+  "include"
+  "interface"
+  "module"
```
Comment (theHamsta) [inline queries/fortran/highlights.scm:42]: this should be probably `@import`

File: queries/fortran/highlights.scm [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+  "cycle"
+  "endfunction"
+  "endmodule"
+  "endprogram"
+  "endsubroutine"
+  "enumerator"
+  "equivalence"
+  "exit"
+  "format"
+  "function"
```
Comment (theHamsta) [inline queries/fortran/highlights.scm:38]: `@keyword.function`

File: queries/fortran/highlights.scm [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+  "include"
+  "interface"
+  "module"
+  "parameter"
+  "print"
+  "program"
+  "read"
+  "return"
+  "stop"
+  "subroutine"
```
Comment (theHamsta) [inline queries/fortran/highlights.scm:49]: `@keyword.function`?

File: queries/fortran/highlights.scm [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+  "use"
+  "write"
+  "implicit"
+  (none)
```
Comment (vigoux) [inline queries/fortran/highlights.scm:55]: Maybe `@constant.builtin`?

Comment (theHamsta) [reply queries/fortran/highlights.scm:55]: I think `none` is a void return type in Fortran

**Comment by theHamsta** [comment]
@argapost could you review?

File: queries/fortran/highlights.scm [hunk 5]
```diff
@@ -0,0 +1,133 @@
...
+  "::"
+  ","
+ ] @punctuation.bracket
+
+(parameters
+  (identifier) @parameter)
+
+(subroutine_statement
+  (name) @function)
+
```
Comment (argapost) [inline queries/fortran/highlights.scm:131]: Is it possible to add here a highlight group for calling the subroutines with:

```
(subroutine_call
	(name) @function)
```

**Review by argapost — APPROVED** [review]
Everything works as expected.

**Review by vigoux — COMMENTED** [review]
Just need to add the changes mentioned and we'll be good to go then!

**Comment by theHamsta** [comment]
There are no changes mentioned. `none` is not a constant, but `implicit none` is put on the top of each file and means don't deduce type based on first letter. `i-l` would mean integer.

**Comment by vigoux** [comment]
There's the `function` keyword that should be highlighted `keyword.function`

**Review by vigoux — APPROVED** [review]

**Pull request merged**